pkg/roachpb/metadata.proto

```diff
@@ -451,3 +451,12 @@ message Version {
   // internal version set to 0.
   optional int32 internal = 4 [(gogoproto.nullable) = false];
 }
+
+// GCRangeHint contains information that MVCC GC could use to optimize data deletion.
+message GCRangeHint {
+  option (gogoproto.equal) = true;
+
+  // estimated_range_removal_time indicates projected time when all data in the range
+  // would become garbage. If empty, no such estimate is known.
+  optional util.hlc.Timestamp estimated_range_removal_time = 2 [(gogoproto.nullable) = false];
```

Comment:
Does this mean that we'd calculate the GC TTL and set the timestamp when the range would fall below the GC TTL? What if the user changes the GC TTL after the drop, to GC it faster?

I think we'll want to set this to the `DeleteRange` evaluation timestamp, and then the GC queue can compare that to the GC TTL.